### PR TITLE
Update to allow access to call objects

### PR DESCRIPTION
**Pull request by michaelbausor**
Supports https://github.com/googleapis/toolkit/pull/1310

### CODE REVIEW

File: src/Testing/MockStubTrait.php [hunk 1]
```diff
@@ -72,7 +73,9 @@ public function _simpleRequest(
             throw new UnderflowException("ran out of responses");
         }
         list($response, $status) = array_shift($this->responses);
-        return new MockUnaryCall($response, $deserialize, $status);
+        $call = new MockUnaryCall($response, $deserialize, $status);
+        $this->callObjects[] = $call;
```
Comment (garrettjonesgoogle) [inline src/Testing/MockStubTrait.php]: I'm not familiar with this syntax, what does it do?

Comment (michaelbausor) [inline src/Testing/MockStubTrait.php]: It appends `$call` to the end of the array (it is preferred over array_push: http://php.net/manual/en/function.array-push.php)

File: src/Testing/MockStubTrait.php [hunk 2]
```diff
@@ -217,6 +226,16 @@ public function getReceivedCallCount()
     }
 
     /**
+     * @return mixed[] The call objects created by calls to the stub
+     */
+    public function getCallObjects()
```
Comment (garrettjonesgoogle) [inline src/Testing/MockStubTrait.php]: this is a mutating call - maybe call it `popCallObjects`

Comment (michaelbausor) [inline src/Testing/MockStubTrait.php]: Updated, along with other instances of mutating calls that used `get`

**Comment by garrettjonesgoogle** [comment]
LGTM

**Comment by michaelbausor** [comment]
Toolkit PR: https://github.com/googleapis/toolkit/pull/1310

**Pull request merged**